### PR TITLE
Cleans up remote content browsing cards

### LINKED ISSUE #10604: Card Cleanup
## Observed behavior
Overall, there could be a little bit of cleanup on some of the new cards on the library page, as there have been some small regressions over the course of the 0.15 patch releases, and some new view more cards have been introduced ([several items from the alpha13 bug bash ](https://www.notion.so/learningequality/0-16-0-alpha13-bug-bash-bdc3ed481a9e4fbbb4fe5d1fdb08668b))

Items that could be tweaked: 
- Would be nice to have some padding or adjust font size on cards in “More” section
 
Screenshot from 2023-04-27 18-43-33

- this is not a real problem (and also I approve the PR related to this) but the spacing of the download button on the content card looks weird
Screenshot 2023-04-27 at 12 45 05 PM

- There’s plenty of space on a page and yet the card is squeezed. We should probably have a min-width and let the cards wrap instead of shrink- between viewport width 850px and 1200px they’re too small 
Screenshot from 2023-04-27 18-45-25

## Context

v0.16.0-a13

### PR DESCRIPTION
**Pull request by akolson**
## Summary

This pr does clean up on cards within the remote content browsing feature.
**Before 1**
image
**After 1**
Device cards are now resized based on available space
Screenshot 2023-05-03 at 20 34 03

**Before 2**
image
**After 2**
Screenshot 2023-05-03 at 20 33 08

**Before 3**
image
**After 3**
Content cards are now resized based on available space
Screenshot 2023-05-03 at 20 32 24

## References

Fixes https://github.com/learningequality/kolibri/issues/10604

## Reviewer guidance

Navigate to the areas in the attached in the screenshots  and try to resize the browser window.

## Testing checklist

- [x] Contributor has fully tested the PR manually
- [x] If there are any front-end changes, before/after screenshots are included
- [ ] Critical user journeys are covered by Gherkin stories
- [ ] Critical and brittle code paths are covered by unit tests

## PR process

- [x] PR has the correct target branch and milestone
- [x] PR has 'needs review' or 'work-in-progress' label
- [x] If PR is ready for review, a reviewer has been added. (Don't use 'Assignees')
- [ ] If this is an important user-facing change, PR or related issue has a 'changelog' label
- [ ] If this includes an internal dependency change, a link to the diff is provided

## Reviewer checklist

- Automated test coverage is satisfactory
- PR is fully functional
- PR has been tested for [accessibility regressions](http://kolibri-dev.readthedocs.io/en/develop/manual_testing.html#accessibility-a11y-testing)
- External dependency files were updated if necessary (`yarn` and `pip`)
- Documentation is updated
- Contributor is in AUTHORS.md

### CODE REVIEW

**Review by thanksameeelian — COMMENTED** [review]
i think "after"s # 1 & # 3 look great and were performing just as intended as i was testing out the PR. it's a huge improvement in appearance and usability!

i'm super curious what @marcellamaki thinks of the before/after of the download icon, but i must confess it still looks a little funky to me (maybe because, for me, the shapes of the two icons make it difficult for them to seem aligned on both top and bottom) 🙂

as an aside, testing this out also made me think it might be nice to have a teeny separate task to responsively handle the "Showing all available libraries around you." text (and maybe delete that errant period, just a personal vendetta haha), as it gets a bit strange during resizing. i think it has only become noticeable now because this PR successfully fixes weirdness with the other visible components. curious what others think!

https://user-images.githubusercontent.com/43046460/236090154-f594805a-1a53-439b-b922-a5c3cee41ef8.mov

**Comment by rtibbles** [comment]
> and maybe delete that errant period, just a personal vendetta haha

https://design-system.learningequality.org/writing/#punctuation

Our design system writing guide does advise against unnecessary periods.

**Comment by akolson** [comment]
Hi @thanksameeelian and @rtibbles for the feedback.
Concerning 2, you might want to use [Grid Ruler](https://chrome.google.com/webstore/detail/grid-ruler/joadogiaiabhmggdifljlpkclnpfncmj?hl=en)(Thanks @marcellamaki, I don't have to strain my eyes anymore :))
Screenshot 2023-05-04 at 09 58 12

Concerning the punctuation, I will just manipulate the translation for now to remove the punctuation without having to make changes to actual string unless this change is acceptable within the string freeze. @rtibbles?

**Comment by akolson** [comment]
@thanksameeelian I have fixed the issues raised particularly your punctuation friend 😃 and the strangeness exhibited by the `Showing all available libraries around you.` string.(see attachment below, Also checkout behavior of this string in the [figma docs](https://www.figma.com/file/MpCG7r5PULOCJsEvnTSti8/Android-Designs-2022?node-id=194-2922)). Very happy for any additional feedback you or anyone else may have regarding this pr 🙏 .

https://user-images.githubusercontent.com/5203639/236154857-69d920c3-038e-4530-a8f7-d77e2922d0d0.mov

**Review by marcellamaki — COMMENTED** [review]
Personally, I think this looks great. The feedback from @thanksameeelian was really meticulous, and you did an excellent job with the final adjustments, @akolson. 

As an aside on the icon buttons... they will sometimes be the bane of your existence, Amelia. Even if they are all "the same size" i.e. the svg is 24px x 24px, sometimes the images themselves are not quite perfectly that size, and will have an extra pixel or two of space _within_ that square svg. Drives me absolutely bonkers. Did I once spend 45 minutes moving an icon up and down and up and down and finally just accepting it was either going to be one pixel too high or one pixel too low, because the actual icon  was an odd number of pixels tall? yes. yes I have. But - grid ruler does help! even if just to identify the fact that the svg is the problem :D I am 99% sure that this is the issue with the download icon, with that tiny amount of "Extra" white space at the bottom.

My last comment before approving is just a final 🆗 from @radinamatic that removing the period from the string using code is acceptable. Radina, the code here is simply removing the final character. I don't have a specific example of a language where this might be a problem, but I just wanted to double check. As an extra confirmation, @akolson, we could always confirm the last character is a "." and in that case remove it, otherwise, not remove it.

**Comment by thanksameeelian** [comment]
@akolson agreed with @marcellamaki  that this looks great!! thank you so much for taking my questions & concerns in stride! :) 

and thank you both for the tips about grid ruler! too often i find myself "knowing" programmatically that two icons are aligned but it simply doesn't look _right_, and this seems like the perfect tool to help release me of that nagging feeling (or come to terms with reality 😅)

**Comment by akolson** [comment]
Check has been effected as requested by Marcella

File: kolibri/plugins/learn/assets/src/views/ChannelCardGroupGrid.vue
Comment (nucleogenesis) [inline kolibri/plugins/learn/assets/src/views/ChannelCardGroupGrid.vue]: The use of numbers here is kind of ambiguous. I had to dive into the `KResponsiveWindow` code to get a grasp on what the numbers refer to and I'm not really sure exactly what these conditions translate to IRL after looking at that code.

It'd be really helpful here if you could add some commentary here explaining what this should translate to on the screen. It strikes me as odd that 6 is lumped in with 0-2 and anything bigger than 6 is lumped in as the smallest possible span.

Comment (nucleogenesis) [reply kolibri/plugins/learn/assets/src/views/ChannelCardGroupGrid.vue]: Also noting there are some other areas with this same logic.

Maybe it would be worth considering moving this logic into a shared parent component to `provide` it and you can `inject` it wherever it is used - rather than adding the same comments across a bunch of files but that'd probably be just fine too I think

**Review by nucleogenesis — CHANGES_REQUESTED** [review]

File: kolibri/plugins/learn/assets/src/views/ChannelCardGroupGrid.vue
Comment (akolson) [reply kolibri/plugins/learn/assets/src/views/ChannelCardGroupGrid.vue]: Hi @nucleogenesis, I have made the changes as recommended. Considering the odd '6', this is intentional and hopefully the comments added to the `layoutSpan` property make it clearer.

File: kolibri/plugins/learn/assets/src/views/LibraryPage/index.vue
Comment (akolson) [inline kolibri/plugins/learn/assets/src/views/LibraryPage/index.vue:453]: Heads up that this is a [work around solution](https://github.com/vuejs/vue/issues/7017#issuecomment-480906691) to using computed properties specified [here](https://vuejs.org/guide/components/provide-inject.html#working-with-reactivity) as adding the configuration failed to work

File: kolibri/plugins/learn/assets/src/views/ExploreLibrariesPage/LibraryItem.vue
Comment (nucleogenesis) [inline kolibri/plugins/learn/assets/src/views/ExploreLibrariesPage/LibraryItem.vue:167]: This explains very well thank you!

File: kolibri/plugins/learn/assets/src/views/LibraryPage/index.vue
Comment (nucleogenesis) [reply kolibri/plugins/learn/assets/src/views/LibraryPage/index.vue:453]: Do you think that maybe to avoid the workaround this could be a `computed()` property in some kind of `useLayout` composable so we can define the `layoutSpan` calculation once? I wonder if there are other places we could simplify with such a composable to have things like "channelCardLayoutSpan", "contentCardLayoutSpan".

Anyway, just a thought for the future, no changes needed in this regard now.

**Review by nucleogenesis — APPROVED** [review]
Code LGTM so should be good to merge if passing manual QA

**Comment by akolson** [comment]
@nucleogenesis I think that would be great. I am just not sure we need this in a composable since it is only scoped to remote content browsing work for now. We might need to perform an audit for areas that would benefit from this.

**Comment by akolson** [comment]
My apologies @radinamatic. I believe I was meant to hold off merging this pending clarification about:

> My last comment before approving is just a final 🆗 from @radinamatic that removing the period from the string using code is acceptable. Radina, the code here is simply removing the final character. I don't have a specific example of a language where this might be a problem, but I just wanted to double check. As an extra confirmation, @akolson, we could always confirm the last character is a "." and in that case remove it, otherwise, not remove it.

Any thoughts?

**Comment by marcellamaki** [comment]
No worries @akolson - I don't think that needed to be a blocker. We can handle in follow up if @radinamatic has concerns :) I also think the updates you made should address it anyways, as you were checking to ensure the character matched before removing it!

**Pull request merged**
